vm/src/warn.rs

```diff
@@ -31,19 +62,130 @@ pub fn warn(
     )
 }
 
+fn get_filter(
+    _category: PyObjectRef,
+    _text: PyObjectRef,
+    _lineno: usize,
+    _module: PyObjectRef,
+    mut _item: PyObjectRef,
+    vm: &VirtualMachine,
+) -> PyResult<()> {
+    let filters = vm.state.warnings.filters.as_object().to_owned();
+
+    let filters: PyListRef = filters
+        .try_into_value(vm)
+        .map_err(|_| vm.new_value_error("_warnings.filters must be a list".to_string()))?;
+
+    /* WarningsState.filters could change while we are iterating over it. */
+    for i in 0..filters.borrow_vec().len() {
+        let tmp_item = filters.borrow_vec().get(i).cloned();
+        let tmp_item = if let Some(tmp_item) = tmp_item {
+            let tmp_item = PyTupleRef::try_from_object(vm, tmp_item)?;
+            if tmp_item.len() != 5 {
+                Err(vm.new_value_error(format!("_warnings.filters item {} isn't a 5-tuple", i)))
+            } else {
+                Ok(tmp_item)
+            }
+        } else {
+            Err(vm.new_value_error(format!("_warnings.filters item {} isn't a 5-tuple", i)))
+        }?;
+
+        /* Python code: action, msg, cat, mod, ln = item */
+        let _action = tmp_item.get(0);
+        let _msg = tmp_item.get(1);
+        let _cat = tmp_item.get(2);
+        let _item_mod = tmp_item.get(3);
+        let _ln_obj = tmp_item.get(4);
+    }
+
+    Ok(())
+}
+
+fn already_warned(
+    registry: PyObjectRef,
+    key: PyObjectRef,
+    should_set: usize,
```

Comment:
```suggestion
    should_set: bool,
```
Maybe trivial, It seems this does not have to be `usize` type.